db/migrations/005.do.epoch-to-timestamp.sql

```diff
@@ -0,0 +1,9 @@
+CREATE OR REPLACE FUNCTION epoch_to_timestamp(epoch INTEGER)
+RETURNS TIMESTAMP AS $$
+DECLARE
+    FILECOIN_GENESIS_UNIX_EPOCH CONSTANT INTEGER := 1598306400;
+BEGIN
+    -- Calculate and return the UNIX timestamp
+    RETURN TO_TIMESTAMP(epoch * 30 + FILECOIN_GENESIS_UNIX_EPOCH);
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
Nice trick to define a stored procedure for converting epochs to timestamps.

Most likely, we will need to add indices (indexes) to speed up the query looking up deals to submit. Do you know if it's possible to index over values calculated using a stored procedure?

Something like this:

```sql
CREATE INDEX active_deals_created_at ON active_deals (epoch_to_timestamp(activated_at_epoch))
```

Maybe should convert timestamps to epochs in the query `WHERE` component, WDYT?

```sql
-- migration script
CREATE INDEX active_deals_activated_at_epoch ON active_deals (activated_at_epoch);
-- query
SELECT 
 -- ...
FROM active_deals
WHERE
  activated_at_epoch < timestamp_to_epoch(NOW() - INTERVAL '2 days')
```

Reply:
Sadly I don't think it's possible to create index over a calculated value like that.

I do like the second option of introducing `timestamp_to_epoch` procedure and adding index to `activated_at_epoch` field.